project-words.txt

```diff
@@ -1,2 +1,11 @@
+chatglm
+endoftext
+fnlp
 llmspec
+logit
+logprobs
 msgspec
+pypa
+PYPI
+pypi
+pytest
```

Comment:
I wonder if we should delete the spell check part. Since there is no lint for it. Maintaining a CI for it may require a huge word dictionary.

Reply:
This file is generated by this command:
```
# install cspell by npm i -g cspell, easy in CI
cspell --words-only --unique '{*.py{,nb},{!({,.}venv)/**/*.{html,py,js,ts,css,md,yaml,yml,json,txt,code-snippets,ipynb,Rmd,R},.github/**/*.{md,yaml,yml}}}' | LC_ALL='C' sort --ignore-case > project-words.txt
```
And is used by the cspell vscode extension to disable warning for legit words that are not in cspell's default dictionary.

I know that not everyone uses the extension, so if you think this makes the repo messier, please feel free to delete it from main branch.

Reply:
Is it generated from the current words in this repo? But how can we know that this is no typo in the current repo?

Reply:
Yes, the command collects unknown words >= 4 chars and saves them to a list.

Most typos would be eliminated if all contributors install the cspell vscode extension or alternative cspell based tools to check and warn about unknown words in the codebase when they're working on it.  
We just need to manually inpsect the collected unknown word list at `project-words.txt` to see if there are typos that are not meant to be there.

Reply:
LGTM